### PR TITLE
ci: Add valgrind run

### LINKED ISSUE #17460: Run tests in memory sanitizer
We run the tests in the thread (and other sanitizers), see `git grep sanitizer ./ci`, but not in the memory sanitizer. This seems like it could be improved. See e.g. #17449

### PR DESCRIPTION
**Pull request by maflcko**
Fixes #17460

### CODE REVIEW

**Comment by MarcoFalke** [comment]
A failing travis build was https://travis-ci.org/bitcoin/bitcoin/builds/623456054

File: ci/test/00_setup_env_native_valgrind.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export LC_ALL=C.UTF-8
+
+export PACKAGES="valgrind clang llvm python3-zmq libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev"
+export USE_VALGRIND=1
+export NO_DEPENDS=1
+export TEST_RUNNER_EXTRA="p2p_segwit.py"  # Only run one test for now. TODO enable all and bump timeouts
+export RUN_FUNCTIONAL_TESTS=true
+export GOAL="install"
+export BITCOIN_CONFIG="--enable-zmq --with-incompatible-bdb --with-gui=no CC=clang CXX=clang++"  # TODO enable GUI
```
Comment (EthanHeilman) [inline ci/test/00_setup_env_native_valgrind.sh]: Should we be testing with this flag `--with-incompatible-bdb` set to true? Is there are reason that we can't use the correct version of berkleydb here?

Comment (MarcoFalke) [reply ci/test/00_setup_env_native_valgrind.sh]: We can use the bdb package from depends or from the Ubuntu package manager. The package manager doesn't come with the bdb version we want, so `--with-incompatible-bdb` is needed. Switching to `./depends` is possible, but shouldn't matter, as we ignore leaks and memory issues in dependencies anyway. They need to be dealt with upstream.

**Comment by MarcoFalke** [comment]
Anything left to do here?

**Comment by practicalswift** [comment]
ACK facb416ad5ba47b01ee52c273a5c5e3709f06f07

Very nice!

**Comment by instagibbs** [comment]
concept ACK!

**Pull request merged**